content/src/main/content/jcr_root/apps/apm/clientlibs/dashboard/js/apm-dashboard.js

```diff
@@ -144,4 +144,103 @@
         return selections[0].items._container.innerHTML.includes("close");
     }
 
-})(window, jQuery);
+    const COMMAND_URL = Granite.HTTP.externalize("/bin/wcmcommand");
+    const deleteText = "Delete";
+    const cancelText = "Cancel";
+
+    $(window).adaptTo("foundation-registry").register("foundation.collection.action.action", {
+        name: "dashboard.delete",
+        handler: function(name, el, config, collection, selections) {
+            const message = createEl("div");
+
+            const intro = createEl("p").appendTo(message);
+            if (selections.length === 1) {
+                intro.text(Granite.I18n.get("You are going to delete the following " + (isFolder(selections) ? "folder:" : "script:")));
+            } else {
+                intro.text(Granite.I18n.get("You are going to delete the following {0} items:", selections.length));
+            }
+
+            const list = [];
+            const maxCount = Math.min(selections.length, 12);

```

Comment:
Why 12 ? Please extract const to avoid using magic numbers or read this value from some config.